lib/CustomFields/pages/EditCustomFieldsRecord/EditCustomFieldsRecord.js

```diff
@@ -0,0 +1,138 @@
+  backendModuleName: PropTypes.string.isRequired,
+  columnCount: PropTypes.number,
+  entityType: PropTypes.string.isRequired,
+  expanded: PropTypes.bool.isRequired,
+  fieldComponent: PropTypes.node.isRequired,
+  okapi: PropTypes.shape({
+    tenant: PropTypes.string.isRequired,
+    token: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+  }).isRequired,
+  onToggle: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  columnCount: 4,
+};
+
+const EditCustomFieldsRecord = ({
+  accordionId,
+  onToggle,
+  expanded,
+  okapi,
+  backendModuleId,
+  backendModuleName,
+  entityType,
+  fieldComponent,
+  columnCount,
+}) => {
+  const {
+    customFields,
+    customFieldsLoaded,
+    customFieldsFetchFailed,
+  } = useCustomFieldsFetch(okapi, backendModuleId, entityType);
+  const {
+    sectionTitle,
+    sectionTitleLoaded,
+    sectionTitleFetchFailed,
+  } = useSectionTitleFetch(okapi, backendModuleName.toUpperCase());
+  const { calloutRef } = useLoadingErrorCallout(customFieldsFetchFailed || sectionTitleFetchFailed);
+  const dataIsLoaded = sectionTitleLoaded && customFieldsLoaded;
+  const customFieldsIsVisible = customFields?.length && customFields?.some(customField => customField.visible);
+  const accordionIsVisible = dataIsLoaded && customFieldsIsVisible;
+  const Field = fieldComponent;
+  const formatedCustomFields = chunk(customFields, columnCount);
+  const columnWidth = rowShapes / columnCount;
+
+  return (
+    <>
+      {accordionIsVisible
+        ? (
+          <Accordion
+            open={expanded}
+            id={accordionId}
+            onToggle={onToggle}
+            label={sectionTitle.value}
+          >
+            {
+              formatedCustomFields.map((row, i) => (
+                <Row key={i}>
+                  {
+                    row.map(customField => (
+                      customField.visible ? (
+                        <Col
+                          data-test-record-edit-custom-field
+                          key={customField.refId}
+                          xs={columnWidth}
+                        >
+                          <Field
+                            component={fieldComponents[customField.type]}
+                            name={`customFields.${customField.refId}`}
+                            required={customField.required}
+                            label={
+                              <>
+                                {customField.name}
+                                {customField.helpText && (
+                                  <InfoPopover
+                                    content={customField.helpText}
+                                    iconSize="small"
+                                  />
+                                )}
+                              </>
+                            }
+                            validate={validateCustomFields(customField)}
+                          />
+                        </Col>
```

Comment:
I would suggest to move it to `renderCustomField` function, but it's not a strong preference. Also, there's no need for a ternary operator. We can simply write `customField.visible && renderCustomField(customField)`